Automattic/jetpack
# Initial State: Replace `isSocialImageGeneratorAvailable` with feature check

## Proposed changes:

* Replace `isSocialImageGeneratorAvailable` with feature check

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* For a Jetpack site with a free plan
    - Go to Social admin page
    - Confirm that you don't see the Social Image Generator (SIG) settings
    - Create a Social Note
    - Confirm that SIG is not visible

* For a Jetpack site with a free plan and on Simple (free/paid) and Atomic sites
    - Create a new post
    - Open Jetpack/Social sidebar
    - Confirm that you don't see SIG option
    - Click Publish to open the pre-publish panel
    - Confirm that you don't see the SIG option
* For a Jetpack site with a paid plan
    - Go to Social admin page
    - Confirm that you DO see the Social Image Generator (SIG) settings
    - Toggle it ON/OFF
    - Confirm that it works fine
    - Create a new post
    - Open Jetpack/Social sidebar
    - Confirm that you DO see SIG option
    - Click Publish to open the pre-publish panel
    - Confirm that you see the SIG option
    - Confirm that the option works fine
    - Create a Social Note
    - Confirm that SIG is not visible
